package-lock.json

```diff
@@ -1,9 +1,12008 @@
 {
   "name": "swarm",
```

Comment:
Do you think it's best to not commit the package-lock.json and just commit the package.json so we can each npm install and generate out own package-lock.json so avoid any issues with it in the future?

Reply:
Totally agree, good call on that!

Reply:
I was poking around this new repo I was subscribed to as part if the BOTSlab organization and saw this PR and thought I would way in here (even though it has already been closed)

It's generally a good idea to commit the `package-lock.json` [as stated in the official docs](https://docs.npmjs.com/cli/v8/configuring-npm/package-lock-json). It provides a snap shot of exactly what versions of dependencies were installed when it was generated, allowing any user (or machine user) to 'time travel' to that exact state.

The reason why this is important is imagine if you had a dep of `packageA: ^1.2.0` which means when running `npm install` it will install the latest version from 1.2.0 to <2.0.0. If user1 runs `npm install` and it installs 1.2.0 as the latest version and they make a change to the repo and commit it. If packageA releases version 1.3.0 with a bug in it, then when user2 checkouts the repo and runs `npm install` they will get version 1.3.0 of packageA which contains the bug which could possibly break the project in the repo. Even though the code for the project in the repo is identical different users will get different behaviour at different times which isn't ideal.

Luckily the people at npm thought of this and provided a command, `npm ci` which install the _exact_ versions as specified in the `package-lock.json` file. This means that user2 in the above scenario would get the same set of dependencies as user1 and everything would work. A user can run `npm install` themselves to get all latest matching version of packages and then after testing that those new versions don't cause any problems (ideally with a bunch of automated tests) they commit the `package-lock.json` that contains the newer versions of the dependencies.

I'm going to @ mention you both here cause I'm not sure how GitHub handles comments on closed PRs. @abigayleh @nshalaby123

Reply:
Okay that's really awesome to know, thank you for telling us this!! We will use this approach going forward!